api/identifiers/views.py

```diff
@@ -55,6 +57,8 @@ class IdentifierList(JSONAPIBaseView, generics.ListAPIView, ListFilterMixin):
 
     permission_classes = (
         IsPublic,
+        EditIfPublic,
```

Comment:
Added EditIfPublic permission class - to only allow the request if it's public.  We only mint DOI's for public resources.